src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/ExternalLanguageOperations.cs

```diff
@@ -0,0 +1,359 @@
+                    }
+                    else
+                    {
+                        ExecuteNonQuery(connection, GetUpdateScript(language, content, parameters, ContentModifyType.Add), parameters);
+
+                    }
+                }
+                foreach (var currentContent in currentLanguage.Contents)
+                {
+                    var content = language.Contents.FirstOrDefault(x => x.PlatformId == currentContent.PlatformId);
+                    if (content == null)
+                    {
+                        ExecuteNonQuery(connection, GetUpdateScript(language, currentContent, parameters, ContentModifyType.Remove), parameters);
+
+                    }
+                }
+            }
+        }
+
+        public virtual void DeleteLanguage(IDbConnection connection, string languageName)
+        {
+            if (string.IsNullOrWhiteSpace(languageName))
+            {
+                throw new LanguageExtensibilityException($"Invalid language name. name: {languageName}");
+            }
+            Dictionary<string, object> parameters = new Dictionary<string, object>();
+            ExecuteNonQuery(connection, GetDropScript(languageName), parameters);
+        }
+
+        /// <summary>
+        /// Returns the status of external languages in a connection
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="languageName"></param>
+        /// <returns></returns>
+        private List<ExternalLanguage> GetLanguages(IDbConnection connection, string languageName = null)
+        {
+            Dictionary<int, ExternalLanguage> dic = new Dictionary<int, ExternalLanguage>();
+            try
+            {
+                using (IDbCommand command = connection.CreateCommand())
+                {
+                    command.CommandText = languageName != null ? GetLanguageScript : GetAllScript;
+                    if (languageName != null)
+                    {
+                        var parameter = command.CreateParameter();
+                        parameter.ParameterName = $"@{LanguageNameParamName}";
+                        parameter.Value = languageName;
+                        command.Parameters.Add(parameter);
+                    }
+                    using (IDataReader reader = command.ExecuteReader())
+                    {
+                        while (reader.Read())
+                        {
+                            int id = reader.GetInt32(0);
+                            string name = reader.GetString(1);
+                            string createdDate = reader.IsDBNull(2) ? string.Empty : reader.GetDateTime(2).ToString();
+                            string owner = reader.IsDBNull(3) ? string.Empty : reader.GetString(3);
+                            string extentionFile = string.Empty;//reader.IsDBNull(4) ? string.Empty : reader.GetString(4);
```

Comment:
Why is this not getting the actual value?

Reply:
removed. the value in db is file bytes. I don't think should be loaded since the file can be too big and can cause perf issues 

---
In reply to: [374368705](https://github.com/microsoft/sqltoolsservice/pull/918#discussion_r374368705) [](ancestors = 374368705)